extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/OpenTelemetryResourceAttrSvcNameTest.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.opentelemetry.deployment;
+
+import static io.opentelemetry.api.trace.SpanKind.SERVER;
+import static io.quarkus.opentelemetry.deployment.common.TestSpanExporter.getSpanByKindAndParentId;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.quarkus.opentelemetry.deployment.common.TestSpanExporter;
+import io.quarkus.opentelemetry.deployment.common.TestSpanExporterProvider;
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+
+public class OpenTelemetryResourceAttrSvcNameTest {
+
+    private static final String SERVICE_NAME = "FrankBullitt";
+
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest().setArchiveProducer(
+            () -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(TestSpanExporter.class)
+                    .addClass(TestSpanExporterProvider.class)
+                    .addAsResource(new StringAsset("" +
+                            "quarkus.otel.bsp.schedule.delay=50\n"), "application.properties"))
+            .overrideRuntimeConfigKey("quarkus.otel.resource.attributes", "service.name=" + SERVICE_NAME);
```

Comment:
We need also a test with properties:
```
quarkus.application.name=app-name
quarkus.otel.service.name=otel-service-name
```
and another one with:
```
quarkus.application.name=app-name
quarkus.otel.service.name=otel-service-name
quarkus.otel.resource.attributes=service.name=otel-resource-service-name
```
To make sure otel-service-name will win.
and another one with:
```
quarkus.application.name=app-name
quarkus.otel.resource.attributes=service.name=otel-resource-service-name
```
To make sure the resource will win.
Maybe creating an abstract test that will just change the properties and maintain the rest of the setup would be more efficient

Reply:
Whether you have application name set or not makes no difference for tests considering resource attribute service name is going to be set anyway.

Reply:
> To make sure the resource will win.

No, the test that I added and sets resource service name tests just that. But as said below, please go ahead with changes.